assets/js/base/utils/address.js

```diff
@@ -21,3 +27,20 @@ export const pluckAddress = ( {
 	city: city.trim(),
 	postcode: postcode ? postcode.replace( ' ', '' ).toUpperCase() : '',
 } );
+
+/**
+ * Sets fields to an empty string in an address if they are hidden by the settings in countryLocale.
+ *
+ * @param {Object} address The address to empty fields from.
+ * @return {Object} The address with hidden fields values removed.
+ */
+export const emptyHiddenAddressFields = ( address ) => {
+	const fields = Object.keys( defaultAddressFields );
+	const addressFields = prepareAddressFields( fields, {}, address.country );
+	addressFields.forEach( ( field ) => {
+		if ( field.hidden ) {
+			address[ field.key ] = '';
```

Comment:
Here you are actually modifying the original address object passed to this function because you are doing operations on the parameter.

The function should return a new object, so you can actually rewrite the function as bellow:

```js
export const emptyHiddenAddressFields = ( address ) => {
	const fields = Object.keys( defaultAddressFields );
	const addressFields = prepareAddressFields( fields, {}, address.country );

	return addressFields.reduce( ( cleanAddress, field ) => {
		cleanAddress[ field.key ] = field.hidden ? '' : address[ field.key ];
		return cleanAddress;
	}, {} );
};
```

Reply:
As discussed in Slack we have implemented a modified version of this. Thanks for catching it!